test/jit/test_list_dict.py

```diff
@@ -408,6 +408,49 @@ def test_over_slice():
             return a[3:10] == [3, 4]
         self.checkScript(test_backward_slice, ())
 
+    def test_slice_index(self):
+        a = torch.tensor(
+            [
+                [[1, 11], [2, 22]],
+                [[3, 33], [4, 44]],
+                [[5, 55], [6, 66]],
+            ]
+        )
+
+        def test_index_slice1(x):
+            x = x[:, :, [0, 1]]
+            return x
+
+        def test_index_slice2(x):
+            x = x[[2, 1, 0], :, :]
+            return x
+
+        def test_index_slice3(x):
+            x = x[[0, 1], :, [1]]
+            return x
+
+        def test_index_slice4(x):
+            x = x[[4], :, :]
+            return x
+
+        def test_index_slice5(x):
+            x = x[[], :, :]
+            return x
+
+        self.checkScript(test_index_slice1, (a,))
```

Comment:
Move check right next to its function under test, ideally all of them should become individual small testcases